language/functional-tests/src/evaluator.rs

```diff
@@ -427,31 +387,17 @@ fn eval_transaction(
                 OutputType::TransactionOutput(txn_output),
             )));
         }
-        ScriptOrModule::Module(parsed_module) => {
-            // stage 2: compile the module
-            if transaction.config.is_stage_disabled(Stage::Compiler) {
-                return Ok(Status::Success);
-            }
-            log.append(EvaluationOutput::Stage(Stage::Compiler));
-
-            let compiled_module =
-                unwrap_or_abort!(compile_module(sender_addr, parsed_module, &*deps)).0;
+        ScriptOrModule::Module(compiled_module) => {
             log.append(EvaluationOutput::Output(Box::new(
                 OutputType::CompiledModule(compiled_module.clone()),
             )));
 
-            // module is added to the list of dependencies despite it passes the verifier or
-            // not
-            deps.push(VerifiedModule::bypass_verifier_DANGEROUS_FOR_TESTING_ONLY(
-                compiled_module.clone(),
-            ));
-
-            // stage 3: verify the module
+            // stage 2: verify the module
             if transaction.config.is_stage_disabled(Stage::Verifier) {
                 return Ok(Status::Success);
             }
             log.append(EvaluationOutput::Stage(Stage::Verifier));
-            let compiled_module = match do_verify_module(compiled_module, &*deps) {
+            let compiled_module = match compiler.verify_module(compiled_module) {
```

Comment:
```
// TODO: Get the dependencies from the fake executor instead of the compiler, or add dependency verification to the transaction execution flow.
```

Reply:
I might be able to do this now, let me take another look